### PR TITLE
[Fix #7902] Change `Link/NumberConversion` to detect symbol form of conversion methods

### LINKED ISSUE #7902: Lint/NumberConversion doesn't handle `to_i` in symbol form
Running `rubocop --only "Lint/NumberConverion" --auto-correct` does not handle the following cases:

```
"1,2,3,foo,5,6,7,8".split(',').map(&:to_i)

"foo".try(:to_i) # when using Rails, returns a 0
```

--------

## Expected behavior

At the minimum, **it should report a violation**.

For auto-correction, I'm not completely sure if this will work for all scenarios, but it could look like the following:
```
"1,2,3,foo,5,6,7,8".split(',').map { |i| Integer(i) } 
```
and
```
"foo".try{ |n| Integer(n) }
```

## Actual behavior
The above violations of using `to_i` are not reported.

## Steps to reproduce the problem

Just use any of the above examples in a file and run rubocop over them.

## RuboCop version
```
$ rubocop -V
0.78.0 (using Parser 2.6.5.0, running on jruby 2.5.3 java)
```

### PR DESCRIPTION
**Pull request by tejasbubane**
Closes #7902

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop-hq/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/lint/number_conversion.rb [hunk 1]
```diff
@@ -52,22 +52,33 @@ class NumberConversion < Base
...
+          (send $_ ${#{METHODS}})
+        PATTERN
+
+        def_node_matcher :to_method_symbol, <<~PATTERN
+          {(send _ $_ ${(sym ${#{METHODS}})} ...)
+           (send _ $_ ${(block_pass (sym ${#{METHODS}}))} ...)}
         PATTERN
 
         def on_send(node)
+          handle_to_method(node)
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/number_conversion.rb:69]: Why not "handle_conversion_method"(as_symbol)? I find the current naming confusing.

File: changelog/change_number_conversion_to_handle_symbols.md [hunk 2]
```diff
@@ -0,0 +1 @@
+* [#7902](https://github.com/rubocop-hq/rubocop/issues/7902): Change `Link/NumberConversion` to detect symbol form of conversion methods. ([@tejasbubane][])
```
Comment (bbatsov) [inline changelog/change_number_conversion_to_handle_symbols.md:1]: Link or Lint? ;-)

File: lib/rubocop/cop/lint/number_conversion.rb [hunk 3]
```diff
@@ -76,13 +87,31 @@ def on_send(node)
...
+        end
 
         def correct_method(node, receiver)
           format(CONVERSION_METHOD_CLASS_MAPPING[node.method_name],
                  number_object: receiver.source)
         end
 
+        def correct_sym_method(to_method)
+          body = format(CONVERSION_METHOD_CLASS_MAPPING[to_method], number_object: 'i')
+          "{ |i| #{body} }"
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/number_conversion.rb:112]: I'd probably suggest here something like `&method(:Integer)` as it's a bit more compact and closer to their original code.

**Comment by bbatsov** [comment]
You also need to update the cop documentation to reflect the extended capabilities of the cop.

File: lib/rubocop/cop/lint/number_conversion.rb (hunk 3 above)
Comment (tejasbubane) [reply lib/rubocop/cop/lint/number_conversion.rb:112]: For `Integer` we pass in base 10 as second argument. We cannot do that with `&method(:Integer)` right?

Comment (tejasbubane) [reply lib/rubocop/cop/lint/number_conversion.rb:118]: Should we use explicit block for `Integer` and method-to-proc for `Float` and `Complex`?

**Comment by tejasbubane** [comment]
@bbatsov Comments resolved - except one comment.

File: lib/rubocop/cop/lint/number_conversion.rb (hunk 3 above)
Comment (bbatsov) [reply lib/rubocop/cop/lint/number_conversion.rb:118]: I guess it'd best to stay uniform with those. You can keep them as blocks.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**